### ISSUE TITLE
3.6.5 - Blockexecution Error: The Scheduled Task XML is Malformed if application title has illegal characters such as &

### ISSUE THREAD

**Issue by CosmicSingularity** [PSAppDeployToolkit/PSAppDeployToolkit#80]
single process for -closeapps 'outlook'. When the script tries to create the task to block execution, In the log I see, error: The task XML is malformed.

**Comment by mmashwani** [comment]
Can you share the full log up to the error you see? Also the full command you're executing? If you can grab the XML file and post that as well, that will help me see what is malformed. The XML should be under this folder I believe: C:\Users\Public\PSAppDeployToolkit

**Comment by CosmicSingularity** [comment]
Command:
```
Show-InstallationWelcome -CloseApps 'Outlook' -BlockExecution -ForceCloseAppsCountdown 7200 -PersistPrompt -TopMost $false -MinimizeWindows $true
```

Log up to the error:
```
[08-19-2015 11:11:26.078] [Initialization] [PSAppDeployToolkit] [Info] :: *******************************************************************************
[08-19-2015 11:11:26.078] [Initialization] [PSAppDeployToolkit] [Info] :: *******************************************************************************
[08-19-2015 11:11:26.245] [Initialization] [PSAppDeployToolkit] [Info] :: [AT&TInc._ConferencingOutlookAdd-in_10.5.15_01] setup started.
[08-19-2015 11:11:26.526] [Initialization] [PSAppDeployToolkit] [Info] :: Script [\\dcmcsccmdell02\source\Software\ATT Connect\AT&T_Conferencing_Outlook_Add-in_v10.5.15\AppDeployToolkit\AppDeployToolkitMain.ps1] dot-source invoked by [\\dcmcsccmdell02\source\Software\ATT Connect\AT&T_Conferencing_Outlook_Add-in_v10.5.15\Deploy-Application.ps1]
[08-19-2015 11:11:26.691] [Initialization] [PSAppDeployToolkitExt] [Info] :: Script [\\dcmcsccmdell02\source\Software\ATT Connect\AT&T_Conferencing_Outlook_Add-in_v10.5.15\AppDeployToolkit\AppDeployToolkitExtensions.ps1] dot-source invoked by [\\dcmcsccmdell02\source\Software\ATT Connect\AT&T_Conferencing_Outlook_Add-in_v10.5.15\AppDeployToolkit\AppDeployToolkitMain.ps1]
[08-19-2015 11:11:26.711] [Initialization] [PSAppDeployToolkit] [Info] :: [AT&TInc._ConferencingOutlookAdd-in_10.5.15_01] script version is [1.0.0]
[08-19-2015 11:11:26.718] [Initialization] [PSAppDeployToolkit] [Info] :: [Deploy Application] script version is [3.6.5]
[08-19-2015 11:11:26.725] [Initialization] [PSAppDeployToolkit] [Info] :: [App Deploy Toolkit Main] script version is [3.6.5]
[08-19-2015 11:11:26.732] [Initialization] [PSAppDeployToolkit] [Info] :: [App Deploy Toolkit Extensions] version is [1.5.0]
[08-19-2015 11:11:26.739] [Initialization] [PSAppDeployToolkit] [Info] :: Computer Name is [DE-WIN7X64-001.DOMAIN.com]
[08-19-2015 11:11:26.746] [Initialization] [PSAppDeployToolkit] [Info] :: Current User is [DOMAIN\ME]
[08-19-2015 11:11:26.753] [Initialization] [PSAppDeployToolkit] [Info] :: OS Version is [Microsoft Windows 7 Enterprise Service Pack 1 64-bit 6.1.7601.65536]
[08-19-2015 11:11:26.760] [Initialization] [PSAppDeployToolkit] [Info] :: OS Type is [Workstation]
[08-19-2015 11:11:26.771] [Initialization] [PSAppDeployToolkit] [Info] :: Current Culture is [en-US] and UI language is [EN]
[08-19-2015 11:11:26.945] [Initialization] [PSAppDeployToolkit] [Info] :: Hardware Platform is [Virtual:VMWare]
[08-19-2015 11:11:26.962] [Initialization] [PSAppDeployToolkit] [Info] :: PowerShell Host is [ConsoleHost] with version [5.0.10105.0]
[08-19-2015 11:11:26.970] [Initialization] [PSAppDeployToolkit] [Info] :: PowerShell Version is [5.0.10105.0 x64]
[08-19-2015 11:11:26.979] [Initialization] [PSAppDeployToolkit] [Info] :: PowerShell CLR (.NET) version is [4.0.30319.34209]
[08-19-2015 11:11:26.986] [Initialization] [PSAppDeployToolkit] [Info] :: *******************************************************************************
[08-19-2015 11:11:27.780] [Initialization] [PSAppDeployToolkit] [Info] :: Display session information for all logged on users: 


NTAccount           : DOMAIN\ME
SID                 : S-1-5-21-3278330371-3405667261-149073299-9931
UserName            : ME
DomainName          : DOMAIN
SessionId           : 1
SessionName         : Console
ConnectState        : Active
IsCurrentSession    : True
IsConsoleSession    : True
IsActiveUserSession : True
IsUserSession       : True
IsRdpSession        : False
IsLocalAdmin        : False
LogonTime           : 8/17/2015 10:55:04 AM
IdleTime            : 00:00:00
DisconnectTime      : 
ClientName          : 
ClientProtocolType  : 
ClientDirectory     : 
ClientBuildNumber   : 0




[08-19-2015 11:11:27.790] [Initialization] [PSAppDeployToolkit] [Info] :: The following users are logged on to the system: [DOMAIN\ME].
[08-19-2015 11:11:27.798] [Initialization] [PSAppDeployToolkit] [Info] :: Current process is running with user account [DOMAIN\ME] under logged in user session for [DOMAIN\ME].
[08-19-2015 11:11:27.806] [Initialization] [PSAppDeployToolkit] [Info] :: The following user is the console user [DOMAIN\ME] (user with control of physical monitor, keyboard, and mouse).
[08-19-2015 11:11:27.814] [Initialization] [PSAppDeployToolkit] [Info] :: The active logged on user is [DOMAIN\ME].
[08-19-2015 11:11:27.821] [Initialization] [PSAppDeployToolkit] [Info] :: The active logged on user [DOMAIN\ME] has a primary UI language of [EN].
[08-19-2015 11:11:27.828] [Initialization] [PSAppDeployToolkit] [Info] :: The following UI messages were imported from the config XML file: [UI_Messages_EN].
[08-19-2015 11:11:27.836] [Initialization] [PSAppDeployToolkit] [Info] :: The active logged on user [DOMAIN\ME] has a DPI scale factor of [100] with DPI pixels [96].
[08-19-2015 11:11:27.984] [Initialization] [PSAppDeployToolkit] [Info] :: Unable to load COM Object [Microsoft.SMS.TSEnvironment]. Therefore, script is not currently running from a SCCM Task Sequence.
[08-19-2015 11:11:27.993] [Initialization] [PSAppDeployToolkit] [Info] :: Skipping attempt to check for and make the task scheduler services healthy because the App Deployment Toolkit is not running under the [NT AUTHORITY\SYSTEM] account.
[08-19-2015 11:11:27.999] [Initialization] [PSAppDeployToolkit] [Info] :: Session 0 not detected.
[08-19-2015 11:11:28.006] [Initialization] [PSAppDeployToolkit] [Info] :: Installation is running in [Interactive] mode.
[08-19-2015 11:11:28.013] [Initialization] [PSAppDeployToolkit] [Info] :: Deployment type is [Installation].
[08-19-2015 11:11:28.094] [Pre-Installation] [Get-RunningProcesses] [Info] :: Check for running application(s) [Outlook]...
[08-19-2015 11:11:28.163] [Pre-Installation] [Get-RunningProcesses] [Info] :: The following processes are running: [OUTLOOK].
[08-19-2015 11:11:28.172] [Pre-Installation] [Get-RunningProcesses] [Info] :: Resolve process descriptions...
[08-19-2015 11:11:28.228] [Pre-Installation] [Get-RunningProcesses] [Info] :: Finished checking running application(s).
[08-19-2015 11:11:28.370] [Pre-Installation] [Show-WelcomePrompt] [Info] :: Prompt user to close application(s) [Microsoft Outlook]...
[08-19-2015 11:11:28.385] [Pre-Installation] [Show-WelcomePrompt] [Info] :: Close applications countdown has [7200] seconds remaining.
[08-19-2015 11:11:28.399] [Pre-Installation] [Show-WelcomePrompt] [Info] :: Close applications countdown has [7200] seconds remaining.
[08-19-2015 11:13:37.196] [Pre-Installation] [Show-InstallationWelcome] [Info] :: User selected to force the application(s) to close...
[08-19-2015 11:13:37.610] [Pre-Installation] [Show-InstallationWelcome] [Info] :: Stop process OUTLOOK...
[08-19-2015 11:13:37.973] [Pre-Installation] [Show-InstallationWelcome] [Info] :: Stop process OUTLOOK...
[08-19-2015 11:13:39.999] [Pre-Installation] [Get-RunningProcesses] [Info] :: Check for running application(s) [Outlook]...
[08-19-2015 11:13:40.019] [Pre-Installation] [Get-RunningProcesses] [Info] :: Application(s) are not running.
[08-19-2015 11:13:40.026] [Pre-Installation] [Get-RunningProcesses] [Info] :: Finished checking running application(s).
[08-19-2015 11:13:40.041] [Pre-Installation] [Show-InstallationWelcome] [Info] :: [-BlockExecution] parameter specified.
[08-19-2015 11:13:40.424] [Pre-Installation] [Block-AppExecution] [Info] :: Create scheduled task to cleanup blocked applications in case installation is interrupted.
[08-19-2015 11:13:40.451] [Pre-Installation] [Get-ScheduledTask] [Info] :: Retrieve Scheduled Tasks...
[08-19-2015 11:13:44.434] [Pre-Installation] [Execute-Process] [Info] :: [C:\Windows\System32\schtasks.exe] is a valid fully qualified path, continue.
[08-19-2015 11:13:44.701] [Pre-Installation] [Execute-Process] [Info] :: Working Directory is [C:\Windows\System32].
[08-19-2015 11:13:44.711] [Pre-Installation] [Execute-Process] [Info] :: Executing [C:\Windows\System32\schtasks.exe /create /f /tn AT&TInc._ConferencingOutlookAdd-in_10.5.15_01_BlockedApps /xml "C:\Users\Public\PSAppDeployToolkit\SchTaskUnBlockApps.xml"]...
[08-19-2015 11:13:44.822] [Pre-Installation] [Execute-Process] [Error] :: Standard error output from the process: ERROR: The task XML is malformed.

(36,235):Arguments:
[08-19-2015 11:13:44.843] [Pre-Installation] [Execute-Process] [Info] :: Execution completed with exit code [1].
[08-19-2015 11:13:44.874] [Pre-Installation] [Block-AppExecution] [Error] :: Failed to create the scheduled task [AT&TInc._ConferencingOutlookAdd-in_10.5.15_01_BlockedApps] by importing the scheduled task XML file [C:\Users\Public\PSAppDeployToolkit\SchTaskUnBlockApps.xml].
```

XML File:
```
<?xml version="1.0" encoding="UTF-16"?>
<Task version="1.2" xmlns="http://schemas.microsoft.com/windows/2004/02/mit/task">
	<RegistrationInfo></RegistrationInfo>
	<Triggers>
		<BootTrigger>
			<Enabled>true</Enabled>
		</BootTrigger>
	</Triggers>
	<Principals>
		<Principal id="Author">
			<UserId>S-1-5-18</UserId>
		</Principal>
	</Principals>
	<Settings>
		<MultipleInstancesPolicy>IgnoreNew</MultipleInstancesPolicy>
		<DisallowStartIfOnBatteries>false</DisallowStartIfOnBatteries>
		<StopIfGoingOnBatteries>false</StopIfGoingOnBatteries>
		<AllowHardTerminate>true</AllowHardTerminate>
		<StartWhenAvailable>false</StartWhenAvailable>
		<RunOnlyIfNetworkAvailable>false</RunOnlyIfNetworkAvailable>
		<IdleSettings>
			<StopOnIdleEnd>false</StopOnIdleEnd>
			<RestartOnIdle>false</RestartOnIdle>
		</IdleSettings>
		<AllowStartOnDemand>true</AllowStartOnDemand>
		<Enabled>true</Enabled>
		<Hidden>false</Hidden>
		<RunOnlyIfIdle>false</RunOnlyIfIdle>
		<WakeToRun>false</WakeToRun>
		<ExecutionTimeLimit>PT1H</ExecutionTimeLimit>
		<Priority>7</Priority>
	</Settings>
	<Actions Context="Author">
		<Exec>
			<Command>powershell.exe</Command>
			<Arguments>-ExecutionPolicy Bypass -NoProfile -NoLogo -WindowStyle Hidden -File "C:\Users\Public\PSAppDeployToolkit\AppDeployToolkitMain.ps1" -CleanupBlockedApps -ReferringApplication "AT&TInc._ConferencingOutlookAdd-in_10.5.15_01"</Arguments>
		</Exec>
	</Actions>
</Task>
```

**Comment by CosmicSingularity** [comment]
Ok, I think I figured it out. I guess, & is an illegal character. I removed the & from the $appVendor variable. 
Before: AT&T Inc. 
After: ATT Inc.

When I removed & from AT&T, didn't get the error.

**Comment by mmashwani** [comment]
You can get the ampersand back in the XML by escaping it with a caret like so: ^&

I will add a fix in the toolkit that will automatically escape the ampersand if one is detected in the string.

**Comment by mmashwani** [comment]
List of special shell characters that need escaping:
http://taskwarrior.org/docs/escapes.html

**Comment by mmashwani** [comment]
This issue should now be resolved in v3.6.6. Please test and let me know if it works as expected now.

**Comment by Manfred538** [comment]
Not working for Arguments in version 3.6.8. Calling a cmd.exe with an argument containing "&" results in "ERROR: The task XML is malformed".
Creating the task in the Task scheduler and exporting it to an XML, changes the "&" to "&amp;" in the arguments section. Think you should change the & to the XML entity    &amp;     if encountered
